plugins/Diagnostics/Diagnostic/Diagnostic.php

```diff
@@ -0,0 +1,22 @@
+<?php
+/**
+ * Piwik - free/libre analytics platform
+ *
+ * @link http://piwik.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ */
+
+namespace Piwik\Plugins\Diagnostics\Diagnostic;
+
+/**
+ * Performs a diagnostic on the system or Piwik.
```

Comment:
> would it be possible to include an example in the docs?

Done.

> Maybe it's important to note that Diagnostics are loaded through DI and thus can use constructor based injection?

I've added a sentence about that in the comment. Maybe we should think that more globally and document it at some point (e.g. controllers, menus, etc.. have DI but it's not documented).

> Also if there are any requirements or conventions implementing classes should follow (like using translations?)

Nothing specific. Regarding translations: the diagnostic results take a label, so it's just a string and the user can use the translator or not. I think we should document when a translation key is expected (because it's a particular case) but not the other way around (because it's the default/normal case).

> For example, a command that lists the types & number of archives in an archive tables (ie, temporary, invalidated, for periods, etc.), number of duplicates if any, number of rows per archive, etc., could be considered a diagnostic tool, but it doesn't perform a simple check. Since these classes do just perform checks, I think it would be good to refer to them as 'diagnostic checks' in documentation.

I'm not sure I understand, these tools cool be implemented as diagnostics too